### DISCUSSION TITLE
First step, hard, in the Frigate world

### DISCUSSION THREAD

**Discussion by ghost** [blakeblackshear/frigate#7691] [General]
Good morning,
In advance, sorry for this long message, which aims to "share" my experience with Frigate.

I discovered Frigate some time ago, and hadn't yet taken the time to look into it, as a possible alternative to "Motion"

It's been 8 days now as I navigate the "meanders" of the config and other considerations

Already, a big thank you to the different developers (I also develop for my part but in other areas) for the great work, hats off.

I quite quickly obtained great results, which were motivating for me, Launching of events: OK, continuous recording: OK, motion detection: OK (great) etc etc. everything was going well.
So I decided to start from scratch after these very promising tests, and that's the drama, on several levels, even very discouraging.

No "reproaches" in my approach, just a different vision from those who must master frigate inside out.
-> the config.yml file: what a hassle to know where there is a misplaced option, the logs certainly generate python errors but it's total fog, impossible to really know where our error is, I'm not talking here about the syntax of YAML files.
there is a "logic" which at the moment escapes me for this or that action, an idiotic example, in the "roles:" the "detect:" allows to specify for example "with/height/ftps..", on the other hand "record:" he no (!)
and "finesses" like that there are lots of, at what level of indentation is this ok or not...etc
and as said above, the logs absolutely don't help, so you have to put everything in comments, restart frigate (see destroy the docker) and test, line by line, it's extremely time-consuming.
The tation document is very complete, really, but it completely hides its "finesses", you have to Google to find user config.yml in the hope of finding a use case.

Currently I don't have anything that works, unless of course I limit myself to very very basic things; and I don't understand why, so I start from zero, again and again /:)

other things, but it's linked to me, I'm aware of it, it's the dockers (first experience), also wow... it's heavy :)
docker compose does not work for me, I have to launch docker frigate via a docker run, is this the correct use?
https://pastebin.com/raw/sk9QPyCD
and launch Portainer.io and company... heavy I said :)

The machine I use is dedicated to video surveillance, under Debian 11, and on this OS I launch another OS (via docker) to run an application: "Frigate" heavy right?, which launches nginx in its docker visibly while I already have Nginx on my debian :)
Question: is it technically possible to "transform" Firgate "docker" version into a classic, standard "application", with the dependencies that go well, of course.. or is it a crazy job?

To benefit from your lights I submit my current config.yml... light: [config.yml](https://pastebin.com/a3CFHkmT)
2 streams from the same cameras,
-> 1 light for detection and events
-> 2 full for recording
Absolutely nothing happens! ... however as indicated before, I managed at the beginning to opt for this "minimum" without worries, I don't see where my mistake is

Thank you for your feedback
PS: I'm not giving up, but you have to take some vitamins :)

_(sorry in advance for my bad English)_

**Comment by NickM-27** [comment]
Regarding the config, frigate has a built in config editor which not only has syntax highlighting and auto suggestion / complete but also won't let you save an invalid config. 

The config error log messages definitely take some time to learn but they certainly provide enough context to fix the issue.

> I have to launch docker frigate via a docker run, is this the correct use?

There's no reason docker compose shouldn't work

> and launch Portainer.io and company... heavy I said :)

Not really a need for that at all, native docker compose is recommended, portainer is known to muck things up

> Question: is it technically possible to "transform" Firgate "docker" version into a classic, standard "application", with the dependencies that go well, of course.. or is it a crazy job?

No this is not supported, nginx is very lightweight also

I don't see the config that you said you're including?

**Reply by NickM-27** [reply]
main problem I see with your config is that you are using tabs (as in the tab character) instead of spaces, this is not recommended. Another reason why using the config editor is recommended

**Reply by ghost** [reply]
you are absolutely right, I use tabs, but they are transformed into spaces... (normally) by my editor
like u see: https://i.goopics.net/lwnjon.jpg

**Reply by NickM-27** [reply]
right, it can be user preference, but IMO 4 spaces is more difficult to read than 2 spaces and that is how the docs are formatted

**Comment by ghost** [comment]
Thank you very much for your quick response

> Regarding the config, frigate has a built in config editor which not only has syntax highlighting and auto suggestion / complete but also won't let you save an invalid config.

In fact I now use this integrated editor from time to time, on the other hand, it is not very talkative about poorly positioned elements (in the structure)

> The config error log messages definitely take some time to learn but they certainly provide enough context to fix the issue.
Is it possible to "externalize" the logs to have them complete, without having to juggle those of Portainer.io or the Frigate interface
in the same approach as folder mounts

> > I have to launch docker frigate via a docker run, is this the correct use?
> 
> There's no reason docker compose shouldn't work
`Docker version 24.0.5, build ced0996`
[docker-compose.yml](https://pastebin.com/raw/6v9nV4Mg)
i just make another try.. `sudo docker compose up` (it's like that?)  it seems ok now (!) https://i.goopics.net/wdc70x.jpg

> > Question: is it technically possible to "transform" Firgate "docker" version into a classic, standard "application", with the dependencies that go well, of course.. or is it a crazy job?
> 
> No this is not supported, nginx is very lightweight also
> 
> I don't see the config that you said you're including?
hups: https://pastebin.com/raw/a3CFHkmT

Thanks a lot

**Reply by NickM-27** [reply]
> it is not very talkative about poorly positioned elements (in the structure)

it will show where errors are and say that element is not meant to be there. 

> i just make another try.. sudo docker compose up (it's like that?) it seems ok now (!) https://i.goopics.net/wdc70x.jpg

yes looks good, you can use `sudo docker compose up -d` to run it in detached mode

at this point, if you have specific errors with the config or questions can answer them

**Reply by ghost** [reply]
Great, thanks to the various leads you gave me, I'm getting "normal" operation again.
Detections are done well, I see the videos in the “events”
the "recording" folder is filling up little by little...
Cool !

However ! ;)
In my simple example (https://pastebin.com/X9bpB9Fr), I used 2 streams from the same camera.
- the first in low resolution/ftps (640x352) for "detect:" detection
- the second in + high resolutions/ftps (1280x720) for recording "record:"

I expected to have the recordings in 1280x720 and here they are in 640x352
  how to define 2 streams from the same camera to detect in low resolution and record in high

because in the end my 1280x720 link is useless since it does not record "either" the detections, "or" a constant recording (even without events)

Reason why I think it was “logical” to define “detect” for the role (1 flow)
```
detect:
             enabled: true
             width: 640
             height: 352
             fps: 15
```

and for the "record" role the same thing:
```
         record:
             enabled: true
             # width: 1280
             # height: 720
             # fps: 25
```

which is obviously not possible.
I must be missing something in the understanding

**Reply by NickM-27** [reply]
The only reason your recordings would be in the wrong resolution is if the camera stream itself is wrong, because that is the correct config. What makes you think that the recordings are the lower resolution?
